app/controllers/TaskController.scala

```diff
@@ -154,13 +162,37 @@ class TaskController @Inject()(annotationDAO: AnnotationDAO,
 
   def duplicateBaseTracings(baseAnnotation: BaseAnnotation, taskParameters: TaskParameters, organizationId: ObjectId)(
       implicit ctx: DBAccessContext,
-      m: MessagesProvider) =
+      m: MessagesProvider) = {
+
+    @SuppressWarnings(Array("TraversableHead")) // We check if nonCancelledTasks are empty before so head always works
+    def checkForTask(taskId: ObjectId): Fox[Annotation] =
+      (for {
+        task <- taskDAO.findOne(taskId)
+        annotations <- annotationDAO.findAllByTaskIdAndType(taskId, AnnotationType.Task)
+      } yield {
+        val nonCancelledTasks = annotations.filter(_.state != AnnotationState.Cancelled)
```

Comment:
I’d rename this to nonCancelledTaskInstances or nonCancelledTaskAnnotations to avoid confusion between task objects and annotation objects :)